drivers/wifi/nrf700x/osal/os_if/inc/osal_api.h

```diff
@@ -321,6 +313,49 @@ void wifi_nrf_osal_spinlock_irq_rel(struct wifi_nrf_osal_priv *opriv,
 				     void *lock,
 				     unsigned long *flags);
 
+
+/**
+ * wifi_nrf_osal_log_dbg() - Log a debug message.
+ * @opriv: Pointer to the OSAL context returned by the @wifi_nrf_osal_init API.
+ * @fmt: Format string.
+ * @...: Variable arguments.
+ *
+ * Logs a debug message.
+ *
+ * Return: Number of characters of the message logged.
+ */
+int wifi_nrf_osal_log_dbg(struct wifi_nrf_osal_priv *opriv,
```

Comment:
Could we keep the previous implementation? This one will make it impossible to compile-out logs based on the configured level and significantly increase the flash usage, right?

Reply:
This doesn't change the core logic, just uses the OS abstraction layers instead of directly calling Zephyr's logging APIs.

Reply:
Yes, that's what I'm saying - this abstraction doesn't allow the compiler to compile-out the logs so I wonder if this is really needed.

Reply:
Yes, that reverting is not a solution, we need to figure out how to get compiler optimisations with keeping the abstractions.

Reply:
Btw, why is the abstraction needed?

Reply:
For 2.1.0 IMHO we should either keep the current implementation which reduces abstraction and flash utilization or revert to the abstract approach but find another way to compile out unnecessary logs. Long story short, we should keep the flash occupancy reduced for the 2.1.0 release. @krish2718 is this what is planned?

Reply:
> Btw, why is the abstraction needed?

`osal` folder is the agnostic folder common across OSes, BUSes, RPU variants etc.

> Long story short, we should keep the flash occupancy reduced for the 2.1.0 release. @krish2718 is this what is planned?

No, the plan is to fix the `osal` logging flash occupancy post-2.1.0, but if this is blocking then I can revert this commit.

Reply:
IMHO the abstraction should allow OSes inject their declarations, not just definitions. I'm not sure if this is blocking, would need to check what is the size of Matter samples after this change, but I think it would be nice if our samples left some spare flash for customers to experiment with their applications, so every kB is precious.

Reply:
Well, that is exactly what is missing in the abstractions, I am fixing the matter build and then I can publish `lock` sample numbers.

Reply:
IMHo, we do have space, so, we can leave this as is and work on it later?

```
Memory region         Used Size  Region Size  %age Used
           FLASH:      839864 B      1016 KB     80.73%
            SRAM:      378494 B       448 KB     82.51%
        IDT_LIST:          0 GB         2 KB      0.00%
```

Reply:
Thanks for sharing the report. Yeah, I think we should be fine with the numbers for now.

Reply:
Reverting that change, slightly increased: 

```
Memory region         Used Size  Region Size  %age Used
           FLASH:      840560 B      1016 KB     80.79%
            SRAM:      378494 B       448 KB     82.51%
        IDT_LIST:          0 GB         2 KB      0.00%

```

Reply:
@Damian-Nordic We should close this comment and investigate further. Do you agree?

Reply:
Ok, these numbers looks good.

Reply:
Yes, I approved the PR already :)